### PR TITLE
#19: move jbrowse config, prep for NPM

### PR DESCRIPTION
**Pull request by samrichca**
- add license
- rename this package to dms-jbrowse-components
- change assembly & default session to props instead of providing them
- export types for assembly, default session, configuration

https://github.com/overture-stack/dms-ui/pull/162

### CODE REVIEW

File: src/components/JbrowseCircular.tsx [hunk 1]
```diff
@@ -43,19 +50,16 @@ export const JbrowseCircular = ({
...
   useEffect(() => {
-    const selectedFileTracks = getTracks(selectedFiles);
+    const selectedFileTracks = getTracks(selectedFiles, assemblyName);
     const state = createViewState({
       ...defaultCircularOptions,
+      ...(defaultSession || {}),
       assembly,
       configuration: { ...defaultCircularOptions.configuration, ...(configuration || {}) },
-      tracks: [...circularTracks, ...selectedFileTracks],
+      tracks: selectedFileTracks,
```
Comment (samrichca) [inline src/components/JbrowseCircular.tsx:59]: remove default tracks for circular view

File: src/components/types.ts [hunk 2]
```diff
@@ -28,3 +28,43 @@ export type JbrowseFileInput = {
 };
 
 export type JbrowseFileFormatDict = Record<JbrowseFileInputFormats, string>;
+
+export type JbrowseAssemblyName = string;
+
+export type JbrowseCircularDefaultSession = {
```
Comment (samrichca) [inline src/components/types.ts:34]: not able to get default session types out of jbrowse. I talked to caroline about it

File: src/components/JbrowseLinear.tsx [hunk 3]
```diff
@@ -21,19 +21,26 @@ import { createViewState, JBrowseLinearGenomeView } from '@jbrowse/react-linear-
...
 import { defaultLinearOptions } from '../utils/linear/dynamic';
 import { getTracks } from './common';
-import { assembly } from '../utils/assembly';
-import { JbrowseFileInput } from './types';
+import { JbrowseFileInput, JbrowseLinearDefaultSession } from './types';
 
-export type LinearViewModel = ReturnType<typeof createViewState>;
+export type JbrowseLinearViewModel = ReturnType<typeof createViewState>;
+export type JbrowseLinearAssembly = JbrowseLinearViewModel['config']['assembly'];
+export type JbrowseLinearConfiguration = JbrowseLinearViewModel['config']['configuration'];
```
Comment (justincorrigible) [inline src/components/JbrowseLinear.tsx:28]: this one needs a trigger warning for me 🤣 
I go "ooooffff"... every. single. time!

File: src/components/types.ts (hunk 2 above)
Comment (justincorrigible) [reply src/components/types.ts:34]: We can potentially assist them in setting that up, if they need a hand?

**Review by justincorrigible — APPROVED** [review]

**Pull request merged**